tests/python/contrib/test_hexagon/conftest.py

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+""" Hexagon testing fixtures used to deduce testing argument
+    values from testing parameters """
+
+import tvm
+from .infrastructure import ceildiv
+
+
+@tvm.testing.fixture
+def shape_nhwc(batch, in_channel, in_size):
+    return (batch, in_size, in_size, in_channel)
+
+
+@tvm.testing.fixture
+def shape_oihw(out_channel, in_channel, kernel):
+    return (out_channel, in_channel, kernel, kernel)
+
+
+@tvm.testing.fixture
+def shape_oihw8i32o4i(out_channel, in_channel, kernel):
+    out_factor, in_factor, in_second_factor = 32, 32, 4
```

Comment:
According to the above comment, should these be 32, 8, 4?

Reply:
These are the factors, not the dimension extents if that helps. 32 input channels are factored out and then those 32 input channels are factored again by a factor of 4 in order to interleave them with the 32 output channels in a 8i32o4i pattern.

Reply:
Resolved by comment above.

Reply:
[Moving my comment out of Thierry's resolved thread.]  I think it would be more clear match the order and numbers in the layout (8, 32, 4) with the factors as follows:
```
in_first_factor, out_factor, in_second_factor = 8, 32, 4
in_factor = in_first_factor * in_second_factor
```